contracts/Nexus.sol

```diff
@@ -72,20 +80,24 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
     ///     - `SIG_VALIDATION_FAILED` (1) denotes signature validation failure allowing simulation calls without a valid signature.
     /// @dev Expects the validator's address to be encoded in the upper 96 bits of the userOp's nonce.
     /// This method forwards the validation task to the extracted validator module address.
+    /// @dev Features Module Enable Mode.
+    /// This Module Enable Mode flow only makes sense for the module that is used as validator
+    /// for the userOp that triggers Module Enable Flow. Otherwise, one should just include
+    /// a call to Nexus.installModule into userOp.callData
     function validateUserOp(
-        PackedUserOperation calldata userOp,
+        PackedUserOperation calldata op,
         bytes32 userOpHash,
         uint256 missingAccountFunds
     ) external virtual payPrefund(missingAccountFunds) onlyEntryPoint returns (uint256 validationData) {
-        address validator;
-        uint256 userOpnonce = userOp.nonce;
-        assembly {
-            validator := shr(96, userOpnonce)
+        PackedUserOperation memory userOp = op;
+        address validator = op.nonce.getValidator();
+        
+        if (op.nonce.isModuleEnableMode()) {
+            userOp.signature = _enableMode(validator, op.signature);
```

Comment:
can be a better name is purpose is to retrieve signature in case of module enable mode.

Reply:
I see it will also install the validator.. but it will verify that deed through only nonce?

Reply:
just looking into the implementation now

Reply:
no, the purpose is to execute the enable flow, signature is just returned to save gas and not parse it again